File: Services/MedicalReport/src/main/java/org/example/medicalreport/Models/entities/MedicalBackground.java
package org.example.medicalreport.Models.entities;

import jakarta.persistence.*;
import lombok.*;
import org.example.medicalreport.Models.enums.*;

import java.time.LocalDate;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
public class MedicalBackground {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id; // Assuming an ID field for the entity
    private FamilialPathology familialPathology;
    private Allergies allergies;
    private MedicalPathology medicalPathology;
    private ChirurgicalPathology chirurgicalPathology;
    private String surgicalApproach;
    private BackgroundType backgroundType;
    private LocalDate date;
    private String comment;
    @ManyToOne
    @JoinColumn(name = "medical_record_id")
    private MedicalRecord medicalRecord;


}

File: front/src/app/Models/MedicalBackground.ts
import { Allergies, ChirurgicalPathology, FamilialPathology, MedicalPathology } from "./enums";

export class MedicalBackground {
  public id!: number | null; // Assuming id can be null for new records
  public familialPathology!: FamilialPathology;
  public allergies!: Allergies;
  public medicalPathology!: MedicalPathology;
  public chirurgicalPathology!: ChirurgicalPathology;
  public surgicalApproach!: string;
  public comment!: string;
  public date!: Date;
  public backgroundType!: string; // Assuming this is a string type for the background type
  public medicalRecordId!: number;

  constructor(
    id: number | null,
    familialPathology: FamilialPathology,
    allergies: Allergies,
    medicalPathology: MedicalPathology,
    chirurgicalPathology: ChirurgicalPathology,
    surgicalApproach: string,
    comment: string,
    date: Date,
    backgroundType: string,
    medicalRecordId: number
  ) {
    this.id = id || null;
    this.familialPathology = familialPathology;
    this.allergies = allergies;
    this.medicalPathology = medicalPathology;
    this.chirurgicalPathology = chirurgicalPathology;
    this.surgicalApproach = surgicalApproach;
    this.comment = comment;
    this.date = date;
    this.backgroundType = backgroundType;
    this.medicalRecordId = medicalRecordId;
  }
}
